spec/features/course/homepage_spec.rb

```diff
@@ -6,10 +6,23 @@
 
   with_tenant(:instance) do
     let(:course) { create(:course, :opened) }
-    let(:achievement_feed_notification) do
-      achievement = create(:course_achievement)
+    let(:feed_notifications) do
```

Comment:
The main issue that me and @allenwq face was that the factories are now somewhat flaky. 

Every time we do a change (adding validations, attributes or doing refactoring to models), we will definitely break something in the factory, and the cause of it might be in the downstream associated models (eg. for `course_assessment_activity_notification`, it could be the `course_user` or `course` that could give problems). Debugging this can be hard too, because sometimes the errors thrown out are at a database level. Because of this, I would imagine that a new member would find it very hard to refactor the factories.

So, we are still figuring out what is the best way forward to allow factories to be easily maintained. I'm not sure what the approaches are, but one of it is to keep the coupling/dependencies between factories light. Unfortunately that results in more boilerplate code in specs, but it is more explicit and maintainable. 

This is also related to #1094 as well. @lowjoel